purchasing/opportunities/front/views.py

```diff
@@ -204,22 +205,31 @@ def signup_for_opp(form, user, opportunity, multi=False):
                 form.errors['opportunities'] = ['That\'s not a valid choice.']
                 return False
             vendor.opportunities.add(_opp)
+            email_opportunities.append(_opp)
     else:
         vendor.opportunities.add(opportunity)
+        email_opportunities.append(opportunity)
 
     if form.data.get('also_categories'):
         # TODO -- add support for categories
         pass
 
     db.session.commit()
+
+    Notification(
```

Comment:
This should probably be gated -- `if opportunity.is_public and opportunity.is_advertised` or something like that.